### PR TITLE
Automated Label Placement

### PR DESCRIPTION
**Pull request by martinvonk**
This pull request adds an option for smart label placement using the [adjustText package](https://github.com/Phlya/adjustText). Currently labels are only placed using an offset but that can cause labels to overlap:
Overzichtskaart_Meetnet_Aamsveen_old_labels
The `adjust_text` function from adjustText uses an iterative adjustment for label positions to minimize overlaps.
Overzichtskaart_Meetnet_Aamsveen_new_labels
In this example the `adjust_text` function is relatively slow (<1min) but in problems with fewer points and shorter label strings it is much faster

This does create an extra dependency but the adjustText package is really small and we already support all it's dependencies.

### CODE REVIEW

File: pastastore/plotting.py [hunk 1]
```diff
@@ -949,21 +963,33 @@ def add_labels(df, ax, **kwargs):
...
-
-        for name, row in df.iterrows():
-            namestr = str(name)
-            txt = ax.annotate(text=namestr,
-                              xy=(row["x"], row["y"]),
-                              fontsize=fontsize,
-                              textcoords=textcoords,
-                              xytext=xytext)
-            txt.set_path_effects(stroke)
+        if adjust:
```
Comment (dbrakenhoff) [inline pastastore/plotting.py:973]: move adjustText import here

File: requirements.txt [hunk 2]
```diff
@@ -3,4 +3,5 @@ pandas>=1.0,<1.2.0
 tqdm>=4.36
 pastas>=0.13.0
 contextily
-pyproj
\ No newline at end of file
+pyproj
+adjustText
```
Comment (dbrakenhoff) [inline requirements.txt:7]: if adjust is tested, then fine to leave it here, otherwise it can be left out here.

File: pastastore/plotting.py [hunk 3]
```diff
@@ -949,21 +963,33 @@ def add_labels(df, ax, **kwargs):
...
+            texts = []
+            for name, row in df.iterrows():
+                texts.append(ax.text(row['x'], row['y'], str(name),
+                                     **{'path_effects': stroke}))
+            adjust_text(texts, force_text=0.05, arrowprops=dict(
```
Comment (dbrakenhoff) [inline pastastore/plotting.py:978]: does adjust_text also take patheffects as a keyword argument? If so would be nice to put it in here, or maybe set them on the resulting annotation labels.

**Review by dbrakenhoff — CHANGES_REQUESTED** [review]
See comments

File: pastastore/plotting.py (hunk 3 above)
Comment (martinvonk) [reply pastastore/plotting.py]: I tried passing the path_effects as both *args and **kwargs to the `adjust_text `function but it does not seem to actually plot it. Passing it to the `ax.text()` currently works so maybe just stick to that?

Comment (dbrakenhoff) [reply pastastore/plotting.py]: As in, passing it to `ax.text` also results in the newly adjusted annotates to have the patheffects as well? In that case, that sounds like a good solution.

File: tests/test_004_maps_plots.py [hunk 4]
```diff
@@ -47,6 +47,11 @@ def test_map_stresses(pstore):
     return
 
 
+def test_map_adjustText(pstore):
```
Comment (dbrakenhoff) [inline tests/test_004_maps_plots.py:50]: you can also set adjust=True in the current plot stresses test. Saves a test.

**Review by dbrakenhoff — COMMENTED** [review]
i would just reuse one of the tests, see comment. After that, looks good!

File: pastastore/plotting.py (hunk 3 above)
Comment (martinvonk) [reply pastastore/plotting.py]: Yes passing with passing it to `ax.text()` you get the result as shown in the second figure of this PR.

**Review by dbrakenhoff — APPROVED** [review]

**Pull request merged**